### PR TITLE
Migrate configure section to 6 0

### PR DESCRIPTION
**Pull request by jonniebigodes**
This pull request addresses the migration of the essentials section to version 6.0 of the docs.

What was done.
- Content was updated with some tweaks for better readability.
- Images and links will need to be further vetted.
- The `toc.js` file was updated to incorporate the changes.

Feel free to provide feedback

### CODE REVIEW

File: docs/configure/overview.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+```js
+// .storybook/main.js
+
+module.exports = {
+  stories: ['../src/**/*.stories.(js|mdx)'],
+  addons: ['@storybook/addon-essentials']
+}
+```
+
+The `main.js` configuration file is a [preset](../presets/introduction) and as such has a powerful interface, but the key fields within it are:
```
Comment (tmeasday) [inline docs/configure/overview.md:23]: I'm not sure we are keeping this link.. perhaps it should link to the preset section of the api/addons section?

File: docs/configure/overview.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+- `stories` - a glob that indicates the [location of your story files](locate-story-files), relative to `main.js`.
```
Comment (tmeasday) [inline docs/configure/overview.md:25]: Next section.

File: docs/configure/overview.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+By default Storybook will load stories from your project based on a glob (pattern matching string) in `.storybook/main.js`. By default that matches all files in your project with extension `.stories.js` -- the intention is you would colocate those files with the component they document.
+
+```
+•
+└── components
+        ├── Button.js
+        └── Button.stories.js
+```
+
+If you want to use a different naming convention, you can alter the glob, using the syntax supported by [micromatch](https://github.com/micromatch/micromatch#extended-globbing).
```
Comment (tmeasday) [inline docs/configure/overview.md:41]: Maybe we should add an example here.

File: docs/configure/integration.md [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+
+Storybook displays your components in a custom web application built using [Webpack](https://webpack.js.org/). Webpack is a complex tool but our default configuration is intended to cover off the majority of use cases. There are also [addons](/addons) available that extend the configuration for other common use cases.
+
+You can customize Storybook's webpack setup by providing a `webpackFinal` field in [`.storybook/main.js`](./overview#configure-your-storybook-project) file.
+
+The value should be an async function that receives a webpack config and eventually returns a webpack config.
+
+#### Default configuration
+
+By default, Storybooks webpack configuration will allow you to:
```
Comment (tmeasday) [inline docs/configure/integration.md:15]: Storybook's

File: docs/configure/integration.md [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+    reactDocgenTypescriptOptions: {
+      shouldExtractLiteralValuesFromEnum: true,
+      propFilter: (prop) => (prop.parent ? !/node_modules/.test(prop.parent.fileName) : true),
+    },
+  },
+};
+```
+
+<div style="background-color:#F8FAFC">
+TODO: vet SEE DOCS link
```
Comment (tmeasday) [inline docs/configure/integration.md:174]: @shilman ?

File: docs/configure/integration.md [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+There are many ways to include CSS in a web application, and correspondingly there are many ways to include CSS in Storybook. Usually it is best to try and replicate what your application does with styling in Storybook’s configuration.
+
+#### CSS-in-JS
+
+CSS-in-JS libraries are designed to use basic JavaScript. They often work in Storybook without any extra configuration. Some libraries expect components to be rendered in a specific rendering “context” (such as to provide themes) and you may need to add a [global decorator](../writing-stories/decorators#global-decorators) to supply it.
+
+#### Importing CSS files
+
+<div style="">
+TODO: check note on sb-storybook-6_0_more_feedback.txt for this
```
Comment (tmeasday) [inline docs/configure/integration.md:197]: Where is this note?

File: docs/configure/integration.md [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+        ```js
+            import '!style-loader!css-loader!./styles.css';
+        ```
+    
+
+To use your CSS in all stories, you simply import it in [`.storybook/preview.js`](./overview#configure-story-rendering)
+
+#### Adding webfonts
+
+If you need webfonts to be available, you may need to add some code to the [`.storybook/preview-head.html`](./story-rendering#adding-to-head) file. We recommend including any assets with your Storybook if possible, in which case you likely want to configure the [static file location](../api/cli-options).
```
Comment (tmeasday) [inline docs/configure/integration.md:215]: We talk about static files just below in this section, I think we link there here

File: docs/configure/user-interface.md [hunk 8]
```diff
@@ -0,0 +1,392 @@
...
+// .storybook/manager.js
+import { addons } from '@storybook/addons';
+import yourTheme from './yourTheme';
+
+addons.setConfig({
+  theme: yourTheme,
+});
+```
+
+The `storybook/theming` package is built using TypeScript, so this should help create a valid theme for typescript users. The types are part of the package itself.
```
Comment (tmeasday) [inline docs/configure/user-interface.md:167]: `@storybook/theming`?

File: docs/configure/user-interface.md [hunk 9]
```diff
@@ -0,0 +1,392 @@
...
+Addons can also hook into the rendering of your story in the preview pane via injecting their own [decorators](../writing-stories/decorators).
+
+Finally, addons can affect the build setup of Storybook by injecting their own webpack configuration to allow the use of other tools in Storybook. Addons that do only this are often referred to as [presets](../presets/introduction).
+
+### Essential, core and community addons
+
+There are many, many Storybook addons, but they can be roughly categorized into three areas:
+
+- Essential addons are core-team developed addons that are considered core to the out-of-the-box user experience of Storybook and thus ship by default with new Storybook installations.
+- Core addons are the addons that are developed by the core team with efforts made to keep them in sync with the development of Storybook itself, and to make sure they are written in idiomatic ways as templates for other addons. They can be found within the Storybook monorepo,
```
Comment (tmeasday) [inline docs/configure/user-interface.md:296]: Link to mono repo?

File: docs/configure/user-interface.md [hunk 10]
```diff
@@ -0,0 +1,392 @@
...
+- Community addons are addons written by the massive Storybook community. They can be found on our website, GitHub, and npm.
```
Comment (tmeasday) [inline docs/configure/user-interface.md:297]: Make our website a link to /addons

File: docs/configure/user-interface.md [hunk 11]
```diff
@@ -0,0 +1,392 @@
...
+
+### Sidebar & URLs
+
+Storybook’s sidebar lists all your stories grouped by component. When you have a lot of components you may wish to also group those components also. To do so, you can add the “/” separator to the “title” of your CSF file and Storybook will group the stories into groups based on common prefixes:
+
+<div style="background-color:#F8FAFC">
+TODO: add image mentioned in the SB 6.0 needs to be further vetted
+</div>
+
+Usually it is a good idea to use a nesting scheme that mirrors the filesystem path of the components. So if you have a file `components/modals/Alert.js` you might name the CSF file `components/modals/Alert.stories.js` and tile it “Components/Modals/Alert”.
```
Comment (tmeasday) [inline docs/configure/user-interface.md:307]: "Title it"

**Review by tmeasday — COMMENTED** [review]
Good stuff, some comments.

File: docs/configure/integration.md (hunk 5 above)
Comment (shilman) [reply docs/configure/integration.md:174]: https://github.com/TypeStrong/fork-ts-checker-webpack-plugin (link 1)
https://github.com/hipstersmoothie/react-docgen-typescript-plugin (link 2)

File: docs/configure/overview.md (hunk 3 above)
Comment (jonniebigodes) [reply docs/configure/overview.md:41]: if you can spare me one. I can gladly add it.

File: docs/configure/user-interface.md (hunk 8 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md:167]: That slipped. Also in the the docs it's also referenced that way :P

File: docs/configure/integration.md (hunk 6 above)
Comment (jonniebigodes) [reply docs/configure/integration.md]: that note slipped through. It was refering to my notes file based on the feedback in the get started section. It's now removed and marking it as solved.

File: docs/configure/integration.md (hunk 7 above)
Comment (jonniebigodes) [reply docs/configure/integration.md]: It's now properly linked.

File: docs/configure/user-interface.md (hunk 9 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: It's linked to the monorepo, more specifically to the addons folder to be more exact.

File: docs/configure/user-interface.md (hunk 10 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Link is added

File: docs/configure/user-interface.md (hunk 11 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: this was another that slipped "2 pairs of eyes" (pardon the bad pun) :P

File: docs/configure/integration.md (hunk 5 above)
Comment (jonniebigodes) [reply docs/configure/integration.md]: @shilman thanks for the links. Already added.

File: docs/configure/integration.md (hunk 4 above)
Comment (jonniebigodes) [reply docs/configure/integration.md]: Fixed with the latest commits.

File: docs/configure/overview.md (hunk 3 above)
Comment (tmeasday) [reply docs/configure/overview.md:41]: Hmm, something like: 

> For example, if you wanted to pull `.md` and `.js` files from the `design-system/src/components` directory, you could write:
> 
> ```
> // .storybook/main.js
> module.exports = {
>   stories: ['../design-system/src/components/*.@{js,md}'],
> };
```

(can you double check my syntax @jonniebigodes? thx)

File: docs/configure/overview.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+module.exports = {
+  stories: ['../src/**/*.stories.(js|mdx)'],
+  addons: ['@storybook/addon-essentials']
+}
+```
+
+The `main.js` configuration file is a [preset](../presets/introduction) and as such has a powerful interface, but the key fields within it are:
+
+- `stories` - a glob that indicates the [location of your story files](#configure-story-loading), relative to `main.js`.
```
Comment (ndelangen) [inline docs/configure/overview.md:25]: it's an array of globs

File: docs/configure/overview.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+If you want to use a different naming convention, you can alter the glob, using the syntax supported by [micromatch](https://github.com/micromatch/micromatch#extended-globbing).
+
+### Configure story rendering
+
+To control the way stories are rendered and add global [decorators](../writing-stories/decorators#global-decorators) and [parameters](..writing-stories/parameters#global-parameters), you can create a  `.storybook/preview.js` file. This is loaded in the Canvas tab, the “preview” iframe that renders your components in isolation. Use `preview.js` for global code (such as [CSS imports](../get-started/setup#render-component-styles) or JavaScript mocks) that applies to all stories.
+
+The `preview.js` file can be an ES module and export the following keys: 
+
+- `decorators` - an array of global [decorators](../writing-stories/decorators#global-decorators)
```
Comment (ndelangen) [inline docs/configure/overview.md:49]: Sometimes we use the word 'list' sometimes 'array'. not super important, but technically 'array' is more correct. Do we use 'list' for a reason?

File: docs/configure/overview.md (hunk 3 above)
Comment (jonniebigodes) [reply docs/configure/overview.md:41]: I've changed it and included the example. According to [this](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md) you were almost right...

I think the correct way to use should be 

```js
module.exports = {
  stories: ['../design-system/src/components/*.@(js|md)'],
};
```

Feel free to correct me if i'm wrong.

File: docs/configure/overview.md (hunk 12 above)
Comment (jonniebigodes) [reply docs/configure/overview.md]: Fixed with the latest commit, thank you for the feedback @ndelangen

File: docs/configure/overview.md (hunk 13 above)
Comment (jonniebigodes) [reply docs/configure/overview.md:59]: @ndelangen it's a styling/semantics thing. It's one of those things that boils down to the situation in question,

File: docs/configure/overview.md (hunk 1 above)
Comment (jonniebigodes) [reply docs/configure/overview.md]: let me know if the link works for you

File: docs/configure/story-rendering.md [hunk 14]
```diff
@@ -0,0 +1,49 @@
+---
+title: 'Story rendering'
+---
+
+In Storybook, your stories render in a special “preview” iframe (Canvas tab) inside the larger Storybook web application. The JavaScript build configuration of the preview is controlled by a [webpack](./integration#default-configuration) config, but you also may want to directly control the HTML that is rendered to help your stories render correctly.
+
+
+### Adding to head html element
+
+If you need to add extra elements to the `head` of the preview, for instance to load static stylesheets, font files, or similar, you can create a file called [`.storybook/preview-head.html`](./overview#configure-story-rendering) and add tags like this:
```
Comment (domyen) [inline docs/configure/story-rendering.md:10]:
```suggestion
If you need to add extra elements to the `<head>` of the preview iframe, for instance to load static stylesheets, font files, or similar, you can create a file called [`.storybook/preview-head.html`](./overview#configure-story-rendering) and add tags like this:
```

File: docs/configure/story-rendering.md [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+<!--  .storybook/preview-head.html -->
+
+<!-- Pull in static files served from your Static director or the internet -->
+<link rel=”preload” href=”your/font” />
+<!-- Or you can load custom head-tag JavaScript: -->
+<script src="https://use.typekit.net/xxxyyy.js"></script>
+<script>try{ Typekit.load(); } catch(e){ }</script>
+```
+
+> Storybook will inject these tags to the iframe where your components are rendered. So, these won’t be loaded into the main Storybook UI.
```
Comment (domyen) [inline docs/configure/story-rendering.md:22]:
```suggestion
> Storybook will inject these tags into the _preview iframe_ where your components are rendered not the Storybook application UI.
```

File: docs/configure/story-rendering.md [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+
+
+### Adding to body html element
+
+Sometimes, you may need to add different tags to the HTML body. This is useful for adding some custom content roots.
```
Comment (domyen) [inline docs/configure/story-rendering.md:27]:
```suggestion
Sometimes, you may need to add different tags to the `<body>`. This is useful for adding some custom content roots.
```

File: docs/configure/story-rendering.md [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+<style>
+  body {
+    font-size: 15px;
+  }
+</style>
+```
+
+> **Important**
+>
+> Storybook will inject these tags to the iframe where your components are rendered. So, these won’t be loaded into the main Storybook UI.
```
Comment (domyen) [inline docs/configure/story-rendering.md:48]:
```suggestion
> Storybook will inject these tags into the _preview iframe_ where your components are rendered not the Storybook application UI.
```

File: docs/configure/story-rendering.md [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+
+```html
+<style>
+  body {
+    font-size: 15px;
+  }
+</style>
+```
+
+> **Important**
```
Comment (domyen) [inline docs/configure/story-rendering.md:46]:
```suggestion
```

File: docs/configure/user-interface.md [hunk 19]
```diff
@@ -0,0 +1,390 @@
...
+| **enableShortcuts**   | Boolean       |Enable/disable shortcuts                                       |`true`                                          |
+| **isToolshown**       | String        |Show/hide tool bar                                             |`true`                                          |
+| **theme**             | Object        |Storybook Theme, see next section                              |`undefined`                                     |
+| **selectedPanel**     | String        |Id to select an addon panel                                    |`my-panel`                                      |
+| **initialActive**     | String        |Select the default active tab on Mobile.                       |`'sidebar'` or `{('sidebar'|'canvas'|'addons')}`|
+| **showRoots**         | Boolean       |Display the top-level grouping as a "root" in the sidebar      |`false`                                         |
+
+### Theming
+
+Storybook is theme-able!
```
Comment (domyen) [inline docs/configure/user-interface.md:44]:
```suggestion
Storybook is theme-able using a lightweight theming API.
```

File: docs/configure/user-interface.md [hunk 20]
```diff
@@ -0,0 +1,390 @@
...
+
+#### Global theming
+
+It's possible to theme Storybook globally.
+
+We've created two basic themes that look good out of the box: "normal" (a light theme) and "dark" (a dark theme). Unless you've set your preferred color scheme as dark Storybook will use the light theme as default.
```
Comment (domyen) [inline docs/configure/user-interface.md:50]:
```suggestion
Storybook includes two themes that look good out of the box: "normal" (a light theme) and "dark" (a dark theme). Unless you've set your preferred color scheme as dark, Storybook will use the light theme as default.
```

File: docs/configure/user-interface.md [hunk 21]
```diff
@@ -0,0 +1,390 @@
...
+// .storybook/manager.js
+import { addons } from '@storybook/addons';
+import yourTheme from './yourTheme';
+
+addons.setConfig({
+  theme: yourTheme,
+});
+```
+
+The `@storybook/theming` package is built using TypeScript, so this should help create a valid theme for typescript users. The types are part of the package itself.
```
Comment (domyen) [inline docs/configure/user-interface.md:167]:
```suggestion
The `@storybook/theming` package is built using TypeScript, so this should help create a valid theme for TypeScript users. The types are part of the package itself.
```

File: docs/configure/user-interface.md [hunk 22]
```diff
@@ -0,0 +1,390 @@
...
+```js
+const Component = styled.div`
+  background: `${props => props.theme.background.app}`
+  width: 0;
+`;
+```
+
+### Storybook addons
+
+A key strength of Storybook is its extensibility. Storybook aims to be your tool of choice for developing components in the browser no matter what choice of UI framework your team has made. Another way that Storybook can fit your team’s development workflow is through addons that allow you to build better and more robust components.
```
Comment (domyen) [inline docs/configure/user-interface.md:275]:
```suggestion
A key strength of Storybook is its extensibility. Use addons to extend and customize Storybook to fit your team’s development workflow.
```

File: docs/configure/user-interface.md [hunk 23]
```diff
@@ -0,0 +1,390 @@
...
+
+Addons have been a part of Storybook since the very beginning and are so integral to the way Storybook works that many of the features described in this documentation are implemented as addons! These addons are installed out of the box with [essentials](../essentials).
```
Comment (domyen) [inline docs/configure/user-interface.md:277]:
```suggestion
Addons are  integral to the way Storybook works. Many of Storybook's core features are implemented as addons! These addons are installed out of the box with [essentials](../essentials). 
```

File: docs/configure/user-interface.md [hunk 24]
```diff
@@ -0,0 +1,390 @@
...
+### Storybook addons
+
+A key strength of Storybook is its extensibility. Storybook aims to be your tool of choice for developing components in the browser no matter what choice of UI framework your team has made. Another way that Storybook can fit your team’s development workflow is through addons that allow you to build better and more robust components.
+
+Addons have been a part of Storybook since the very beginning and are so integral to the way Storybook works that many of the features described in this documentation are implemented as addons! These addons are installed out of the box with [essentials](../essentials). 
+
+#### Addon features
+
+<div>
+todo:vet links here
```
Comment (domyen) [inline docs/configure/user-interface.md:282]: These aren't links

File: docs/configure/user-interface.md [hunk 25]
```diff
@@ -0,0 +1,390 @@
...
+</div>
+
+The most obvious thing addons affect in Storybook is the UI of Storybook itself. Within the UI the [toolbar]() and [addons panel]() are the two chief places addons will appear.
```
Comment (domyen) [inline docs/configure/user-interface.md:285]:
```suggestion
The most obvious thing addons affect in Storybook is the UI of Storybook itself. Within the UI the **toolbar** and **addons panel** are the two chief places addons will appear. 
```

File: docs/configure/user-interface.md [hunk 26]
```diff
@@ -0,0 +1,390 @@
...
+
+Addons can also hook into the rendering of your story in the preview pane via injecting their own [decorators](../writing-stories/decorators).
+
+Finally, addons can affect the build setup of Storybook by injecting their own webpack configuration to allow the use of other tools in Storybook. Addons that do only this are often referred to as [presets](../presets/introduction).
+
+### Essential, core and community addons
+
+There are many, many Storybook addons, but they can be roughly categorized into three areas:
+
+- Essential addons are core-team developed addons that are considered core to the out-of-the-box user experience of Storybook and thus ship by default with new Storybook installations.
```
Comment (domyen) [inline docs/configure/user-interface.md:297]:
```suggestion
- **Essential** addons are core-team developed addons that are considered a part of the out-of-the-box user experience. These ship by default with new Storybook installations.
```

File: docs/configure/user-interface.md [hunk 27]
```diff
@@ -0,0 +1,390 @@
...
+- Core addons are the addons that are developed by the core team with efforts made to keep them in sync with the development of Storybook itself, and to make sure they are written in idiomatic ways as templates for other addons. They can be found within the [Storybook monorepo](https://github.com/storybookjs/storybook/tree/next/addons).
```
Comment (domyen) [inline docs/configure/user-interface.md:298]:
```suggestion
- **Core** addons are developed by the core team. They are kept in sync with the development of Storybook itself and written in idiomatic ways as templates for other addons. They can be found within the [Storybook monorepo](https://github.com/storybookjs/storybook/tree/next/addons).
```

File: docs/configure/user-interface.md [hunk 28]
```diff
@@ -0,0 +1,390 @@
...
+- Community addons are addons written by the massive Storybook community. They can be found on our [website](/addons), [GitHub](https://github.com/), and [npm](https://www.npmjs.com/).
+
+### Sidebar & URLs
+
+Storybook’s sidebar lists all your stories grouped by component. When you have a lot of components you may wish to also group those components also. To do so, you can add the “/” separator to the “title” of your CSF file and Storybook will group the stories into groups based on common prefixes:
```
Comment (domyen) [inline docs/configure/user-interface.md:303]:
```suggestion
Storybook’s sidebar lists all your stories grouped by component. When you have a lot of components you may wish to also group those components also. To do so, you can add the `/` separator to the `title` of your CSF file and Storybook will group the stories into groups based on common prefixes:
```

File: docs/configure/user-interface.md [hunk 29]
```diff
@@ -0,0 +1,390 @@
...
+
+![Storybook sidebar anatomy](./sidebar-anatomy.jpg)
+
+Usually it is a good idea to use a nesting scheme that mirrors the filesystem path of the components. So if you have a file `components/modals/Alert.js` you might name the CSF file `components/modals/Alert.stories.js` and title it “Components/Modals/Alert”.
```
Comment (domyen) [inline docs/configure/user-interface.md:307]:
```suggestion
We recommend using a nesting scheme that mirrors the filesystem path of the components. For example, if you have a file `components/modals/Alert.js` name the CSF file `components/modals/Alert.stories.js` and title it `Components/Modals/Alert`.
```

File: docs/configure/user-interface.md [hunk 30]
```diff
@@ -0,0 +1,390 @@
...
+
+#### Roots
+
+By default Storybook will treat your highest level of groups as “roots”--which are displayed in the UI as “sections” of the hierarchy. Lower level groups are displayed as expandable items in the hierarchy:
```
Comment (domyen) [inline docs/configure/user-interface.md:311]:
```suggestion
By default, Storybook will treat your highest level of groups as “roots”--which are displayed in the UI as “sections” of the hierarchy. Lower level groups are displayed as expandable items in the hierarchy:
```

File: docs/configure/user-interface.md [hunk 31]
```diff
@@ -0,0 +1,390 @@
...
+
+![Storybook sidebar story roots](./sidebar-roots.jpg)
+
+If you’d prefer all groups to be expandable, you can set the showRoots option to false in  [`./storybook/manager.js`](./overview#configure-story-rendering):
```
Comment (domyen) [inline docs/configure/user-interface.md:315]:
```suggestion
If you’d prefer all groups to be expandable, you can set the `showRoots` option to `false` in  [`./storybook/manager.js`](./overview#configure-story-rendering):
```

File: docs/configure/user-interface.md [hunk 32]
```diff
@@ -0,0 +1,390 @@
...
+```
+
+Then we can access these environment variables anywhere inside our preview JavaScript code like below:
+
+```js
+console.log(process.env.STORYBOOK_THEME);
+console.log(process.env.STORYBOOK_DATA_KEY);
+```
+
+You can also access these variables in your custom head/body (see below) using the substitution `%STORYBOOK_X%`, for example: `%STORYBOOK_THEME%` will become `red`.
```
Comment (domyen) [inline docs/configure/user-interface.md:385]:
```suggestion
You can also access these variables in your custom `<head>`/`<body>` (see below) using the substitution `%STORYBOOK_X%`, for example: `%STORYBOOK_THEME%` will become `red`.
```

**Review by domyen — CHANGES_REQUESTED** [review]

File: docs/configure/integration.md [hunk 33]
```diff
@@ -0,0 +1,307 @@
...
+Storybook has built-in Typescript support, so your Typescript project should work with zero configuration needed.
+
+#### Default configuration
+
+The base Typescript configuration uses [`babel-loader`](https://webpack.js.org/loaders/babel-loader/) for Typescript transpilation, and optionally [`fork-ts-checker-webpack-plugin`](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin) for checking.
+
+Each framework uses the base configuration unless otherwise specified:
+
+- Angular ignores the base and uses `ts-loader` and `ngx-template-loader`.
+- Vue ignores the uses `ts-loader` and applies it to both `.tsx` and `.vue` files.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/configure/integration.md]: Fixed with the latest commit

File: docs/configure/integration.md [hunk 34]
```diff
@@ -0,0 +1,307 @@
...
+
+
+
+### Styling and CSS
+
+There are many ways to include CSS in a web application, and correspondingly there are many ways to include CSS in Storybook. Usually it is best to try and replicate what your application does with styling in Storybook’s configuration.
+
+#### CSS-in-JS
+
+CSS-in-JS libraries are designed to use basic JavaScript. They often work in Storybook without any extra configuration. Some libraries expect components to be rendered in a specific rendering “context” (such as to provide themes) and you may need to add a [global decorator](../writing-stories/decorators#global-decorators) to supply it.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/configure/integration.md]: Fixed with the latest commit.

File: docs/configure/overview.md [hunk 35]
```diff
@@ -0,0 +1,69 @@
...
+// .storybook/main.js
+
+module.exports = {
+  stories: ['../my-project/src/components/*.@(js|md)'],
+};
+```
+
+### Configure story rendering
+
+To control the way stories are rendered and add global [decorators](../writing-stories/decorators#global-decorators) and [parameters](..writing-stories/parameters#global-parameters), you can create a  `.storybook/preview.js` file. This is loaded in the Canvas tab, the “preview” iframe that renders your components in isolation. Use `preview.js` for global code (such as [CSS imports](../get-started/setup#render-component-styles) or JavaScript mocks) that applies to all stories.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/configure/overview.md]: Fixed with the latest commit

File: docs/configure/user-interface.md (hunk 28 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Fixed with the latest commit

File: docs/configure/user-interface.md (hunk 29 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Fixed with the latest commit

File: docs/configure/user-interface.md (hunk 31 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Fixed with the latest commit

File: docs/configure/user-interface.md (hunk 32 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Fixed with the latest commit.

File: docs/configure/user-interface.md (hunk 30 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Fixed with the latest commit.

File: docs/configure/story-rendering.md (hunk 15 above)
Comment (jonniebigodes) [reply docs/configure/story-rendering.md]: Fixed with the latest commit.

File: docs/configure/story-rendering.md (hunk 16 above)
Comment (jonniebigodes) [reply docs/configure/story-rendering.md]: Fixed with the latest commit.

File: docs/configure/story-rendering.md (hunk 17 above)
Comment (jonniebigodes) [reply docs/configure/story-rendering.md]: Fixed with the latest commit.

File: docs/configure/story-rendering.md (hunk 18 above)
Comment (jonniebigodes) [reply docs/configure/story-rendering.md]: Removed with the latest commit.

File: docs/configure/user-interface.md (hunk 20 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Fixed with the latest commit.

File: docs/configure/user-interface.md (hunk 22 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Fixed with the latest commit.

File: docs/configure/user-interface.md (hunk 23 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Fixed with the latest commits.

File: docs/configure/user-interface.md (hunk 25 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Fixed with latest commits.

File: docs/configure/user-interface.md (hunk 26 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Fixed with the latest commits

File: docs/configure/user-interface.md (hunk 27 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Fixed with the latest commits.

File: docs/configure/user-interface.md (hunk 24 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md:282]: Fixed with the latest commits.

File: docs/configure/user-interface.md (hunk 21 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Fixed with latest commits.

File: docs/configure/user-interface.md (hunk 19 above)
Comment (jonniebigodes) [reply docs/configure/user-interface.md]: Fixed with latest commits.

File: docs/configure/overview.md [hunk 36]
```diff
@@ -0,0 +1,69 @@
...
+The `main.js` configuration file is a [preset](../api/addons#addon-presets) and as such has a powerful interface, but the key fields within it are:
+
+- `stories` - a array of globs that indicates the [location of your story files](#configure-story-loading), relative to `main.js`.
+- `addons` - a list of the [addons](/addons) you are using.
+- `webpackFinal` - custom [webpack configuration](./integration#extending-storybooks-webpack-config).
+- `babel` - custom [babel configuration](./integration#babel).
+
+### Configure story loading
+
+By default Storybook will load stories from your project based on a glob (pattern matching string) in `.storybook/main.js`. By default that matches all files in your project with extension `.stories.js` -- the intention is you would colocate those files with the component they document.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/configure/overview.md]: fixed with latest commits.

File: docs/configure/story-rendering.md (hunk 14 above)
Comment (jonniebigodes) [reply docs/configure/story-rendering.md]: Fixed with latest commits.

**Comment by RichardVoyles09** [comment]
Thank you

**Review by domyen — APPROVED** [review]

**Comment by jonniebigodes** [comment]
Merging as is. Further improvements will be made into subsequent pull requests.

**Pull request merged**